nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -0,0 +1,178 @@
+            "    static interface NullableTypeParamInterface<E extends @Nullable Object>{}",
+            "    // BUG: Diagnostic contains: Generic type parameter",
+            "    static class InvalidInterfaceImplementation implements NonNullTypeParamInterface<@Nullable String> {}",
+            "    static class ValidInterfaceImplementation implements NullableTypeParamInterface<String> {}",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void nestedTypeParams() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.nullness.Nullable;",
+            "class Test {",
+            "    static class NonNullTypeParam<E> {}",
+            "    static class NullableTypeParam<E extends @Nullable Object> {}",
+            "    // BUG: Diagnostic contains: Generic type parameter",
+            "    static void testBadNonNull(NullableTypeParam<NonNullTypeParam<@Nullable String>> t) {",
+            "        // BUG: Diagnostic contains: Generic type parameter",
+            "        NullableTypeParam<NonNullTypeParam<NonNullTypeParam<@Nullable String>>> t2 = null;",
+            "        // BUG: Diagnostic contains: Generic type parameter",
+            "        t2 = new NullableTypeParam<NonNullTypeParam<NonNullTypeParam<@Nullable String>>>();",
+            "        // this is fine",
+            "        NullableTypeParam<NonNullTypeParam<NullableTypeParam<@Nullable String>>> t3 = null;",
+            "    }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void returnTypeParamInstantiation() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.nullness.Nullable;",
+            "class Test {",
+            "    static class NonNullTypeParam<E> {}",
+            "    static class NullableTypeParam<E extends @Nullable Object> {}",
+            "    // BUG: Diagnostic contains: Generic type parameter",
+            "    static NonNullTypeParam<@Nullable String> testBadNonNull(NonNullTypeParam<String> t) {",
+            "          return t;",
+            "    }",
+            "    static NullableTypeParam<@Nullable String> testOKNull() {",
+            "          return new NullableTypeParam<@Nullable String>();",
+            "    }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void testOKNewClassInstantiationForOtherAnnotations() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import lombok.NonNull;",
+            "class Test {",
+            " static class NonNullTypeParam<E> {}",
+            " static void testBadNonNull(NonNullTypeParam<String> t) {",
+            "        // should not show error for annotation other than @Nullable",
+            "        testBadNonNull(new NonNullTypeParam<@NonNull String>());",
```

Comment:
Should also test the other annotation being in the upper bound like `T extends @Nonnull Object`